src/test/java/org/terasology/itempipes/ItemPipesTest.java

```diff
@@ -28,91 +32,105 @@
-import org.terasology.moduletestingenvironment.ModuleTestingEnvironment;
+import org.terasology.moduletestingenvironment.MTEExtension;
+import org.terasology.moduletestingenvironment.ModuleTestingHelper;
+import org.terasology.moduletestingenvironment.extension.Dependencies;
+import org.terasology.moduletestingenvironment.extension.UseWorldGenerator;
+import org.terasology.registry.In;
 import org.terasology.world.BlockEntityRegistry;
 import org.terasology.world.WorldProvider;
 import org.terasology.world.block.Block;
 import org.terasology.world.block.BlockManager;
-import org.terasology.world.block.entity.placement.PlaceBlocks;
+import org.terasology.world.block.BlockRegion;
 import org.terasology.world.block.family.BlockFamily;
+import org.terasology.world.block.family.BlockPlacementData;
 import org.terasology.world.block.items.BlockItemFactory;
-import org.terasology.world.block.items.OnBlockItemPlaced;
 
 import java.util.EnumSet;
-import java.util.Set;
 
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
-public class ItemPipesTest extends ModuleTestingEnvironment {
+@ExtendWith(MTEExtension.class)
+@UseWorldGenerator("ModuleTestingEnvironment:empty")
+@Dependencies({"ItemPipes", "CoreAdvancedAssets"})
+public class ItemPipesTest {
+    @In
     private WorldProvider worldProvider;
+    @In
     private BlockManager blockManager;
+    @In
     private BlockEntityRegistry blockEntityRegistry;
+    @In
     private EntityManager entityManager;
+    @In
     private PipeSystem pipeSystem;
+    @In
     private Time time;
+    @In
+    private ModuleTestingHelper helper;
 
-    @Override
-    public Set<String> getDependencies() {
-        Set<String> modules = Sets.newHashSet();
-        modules.add("engine");
-        modules.add("CoreAdvancedAssets");
-        modules.add("SegmentedPaths");
-        modules.add("ItemPipes");
-        return modules;
-    }
+    private BlockFamily itemPipesBlockFamily;
+    private BlockFamily chestFamily;
+    private Block airBlock;
 
-    @Before
+    @BeforeEach
     public void initialize() {
-        worldProvider = getHostContext().get(WorldProvider.class);
-        blockManager = getHostContext().get(BlockManager.class);
-        blockEntityRegistry = getHostContext().get(BlockEntityRegistry.class);
-        entityManager = getHostContext().get(EntityManager.class);
-        pipeSystem = getHostContext().get(PipeSystem.class);
-        time = getHostContext().get(Time.class);
+        airBlock = blockManager.getBlock("engine:air");
+        itemPipesBlockFamily = blockManager.getBlockFamily("ItemPipes:basicPipe");
+        chestFamily = blockManager.getBlockFamily("CoreAdvancedAssets:Chest.LEFT");
+
+        BlockRegion region = new BlockRegion(0, 0, 0).expand(5, 5, 5);
+        for (Vector3ic pos : region) {
+            helper.forceAndWaitForGeneration(JomlUtil.from(pos));
+            worldProvider.setBlock(pos, airBlock);
+        }
```

Comment:
as mentioned on Discord, I very much support of having a test API that doesn't need this.

and on a more thorough reading I'm a little surprised that it's necessary to have this **and** the `forceAndWaitForGeneration` calls in the `placePipe` helper?

but this isn't an MTE PR, so if you've been through all this before and this is what it takes to make it work, I'd believe it.